### PR TITLE
Improve Maintainability as suggested by CodeClimate

### PR DESCRIPTION
**Pull request by laceysanderson**
# Tripal 4 Core Dev Task

Issue #1404 
*Issue not completely fixed by this PR*

## Description

1. Adjusted the limits used to measure maintainability for our project

Specifically,

- I expanded the max argument count to 8 since Drupal typically has a lot of arguments for dependency injection purposes and what not.
- I expanded the method complexity substantially as the limit felt unobtainable.
- I turned off the identical code detection as it was detecting service calls which should be identical and thus providing many false positives.

2. Expanded some if clauses with way too many criteria for better readability.

Specifically, I updated two identical if clauses in the ChadoSchema classes (both deprecated API class and new Tripal DBX ChadoSchema class). As you will see from the diff I basically just expanded a single if with way too many arguments to multiple if statements which spread out those same arguments. I feel this is much more readable and will be easier to debug --code climate agrees.

## Testing?
The changes to the CodeClimate yml do not need testing as they are just settings.
The ChadoSchema change should be caught by automated testing.
You can see the change in maintainability issues here if you are interested: https://codeclimate.com/github/tripal/tripal/compare/tv4g0-issue1404-improveMaintainability

So basically this PR just needs someone to ok the change in settings as reasonable :-)

### CODE REVIEW

**Review by Ferrisx4 — APPROVED** [review]
These changes seem to make sense given the nature of Tripal.

**Review by spficklin — APPROVED** [review]
